semantic_release/repository.py

```diff
@@ -0,0 +1,62 @@
+"""Helper for using Twine to upload to an artifact repository.
+"""
+import logging
+import os
+from typing import List
+
+from invoke import run
+
+from semantic_release import ImproperConfigurationError
+
+from semantic_release.settings import config
+
+from .helpers import LoggedFunction
+
+logger = logging.getLogger(__name__)
+
+
+class Repository(object):
+
+    def __init__(self, username: str = None, password: str = None) -> None:
```

Comment:
Make these not optional, having the default spread between this and the get_repository method can be confusing.